### DISCUSSION TITLE
Clarify definition of “whitespace” / “blank characters”

### DISCUSSION THREAD

**Discussion by vladdeSV** [jotaen/klog#375] [Feature ideas & discussions]
**tl;dr**: in the whitespace definition, replace `another character` with `any other characters`.

---

In the specification, “whitespace” is defined as:
> A “space”, a “tab”, or another character that appears blank

I interpret the "*[…] or another character […]*" as "*[…] or **one other** character […]*", as if there were three distinct whitespace characters. The last one mentioned is not specified, and thus could be interpreted as letting the user select a third blank character, or possibly that the third character was forgotten and not put in the specification.

A more appropriate definition would be:
> A “space”, a “tab”, or any other characters that appears blank

**Comment by vladdeSV** [comment]
And I would like to mention (even though this is overkill), with a custom typeface, any character *could* appear blank. Eg. if I make my own font "vladde.ttf" and made the letter `a` appear as a blank character, this specification would allow it.

However, going this far is overkill, in my opinion.

**Comment by jotaen** [comment]
> tl;dr: in the whitespace definition, replace another character with any other characters.

Good point, that should be fixed. I’m not determined yet what protocol to follow regarding spec changes now that we track versions, because I don’t just want to commit to master and cut a new version for every tiny change. I’ll come up with a procedure.

> And I would like to mention (even though this is overkill), with a custom typeface, any character could appear blank.

Appreciate the nit-pick, you are spot on actually. This phrasing was intentionally sloppy, not for a good reason but more out of laziness back then. Since the spec refers to unicode character classes for what a “letter” is, I think it’s best to base the definition of whitespace on that as well. (See https://en.wikipedia.org/wiki/Whitespace_character#Unicode).

**Comment by jotaen** [comment]
Also, linebreaks are technically whitespace as well, but that’s not how the term is used in the spec.

**Comment by jotaen** [comment]
Fixed via https://github.com/jotaen/klog/pull/119.
